main.sh

```diff
@@ -153,6 +153,9 @@ fi
 for tool in "${tools[@]}"; do
     if [[ "${tool}" == *"@"* ]]; then
         version="${tool#*@}"
+        if [[ ! "${version}" =~ ^[1-9][0-9]*$|^[0-9]+\.[1-9][0-9]*$|^[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z\.-]+)?(\+[0-9A-Za-z\.-]+)?$|^latest$ ]]; then
```

Comment:
Sorry, I noticed that the code I previously suggested could not handle a version like `1.0`...

```suggestion
        if [[ ! "${version}" =~ ^([1-9][0-9]*(\.[0-9]+(\.[0-9]+)?)?|0\.[1-9][0-9]*(\.[0-9]+)?|^0\.0\.[0-9]+)(-[0-9A-Za-z\.-]+)?(\+[0-9A-Za-z\.-]+)?$|^latest$ ]]; then
```

Reply:
It's getting so complicated so quickly...

Maybe we should rewrite in python?
Then we can also use batch installation mode of `cargo-binstall`.